### PR TITLE
Report

### PR DESCRIPTION
**Pull request by Sjmarf**

### CODE REVIEW

File: Sources/MlemMiddleware/API Client/ApiClient+Report.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            page: page,
+            limit: limit,
+            unresolvedOnly: unresolvedOnly
+        )
+        let response = try await perform(request)
+        return await caches.report.getModels(api: self, from: response.privateMessageReports)
+    }
+    
+    @discardableResult
+    func resolveReport(
```
Comment (EricBAndrews) [inline Sources/MlemMiddleware/API Client/ApiClient+Report.swift]: Do we want to have this broken out into separate functions for each report type? We define the `Report` abstraction as a wrapper on top of the Lemmy API, so it makes more sense to me for the abstraction work to happen in `Report` and keep `ApiClient` dumb and closely mapped to the Lemmy API.

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Looks good! One refactoring note.

File: Sources/MlemMiddleware/API Client/ApiClient+Report.swift (hunk 1 above)
Comment (Sjmarf) [reply Sources/MlemMiddleware/API Client/ApiClient+Report.swift]: Sure, that makes sense 👍

**Review by EricBAndrews — APPROVED** [review]
LGTM! 🔥

**Pull request merged**